native/src/main/java/io/ballerina/stdlib/http/api/nativeimpl/ExternResponseProcessor.java

```diff
@@ -0,0 +1,522 @@
+                    StringUtils.fromString(contentType) : null, mediaTypeType);
+            return validateConstraints(requireValidation, convertedValue, mediaTypeType, "media-type binding failed");
+        } catch (BError conversionError) {
+            throw HttpUtil.createHttpError("media-type binding failed", CLIENT_ERROR, conversionError);
+        }
+    }
+
+    private static String getContentType(BObject response) {
+        HttpHeaders httpHeaders = (HttpHeaders) response.getNativeData(HTTP_HEADERS);
+        return httpHeaders.get("Content-Type");
+    }
+
+    private static Object getHeadersMap(BObject response, Type headersType, boolean requireValidation) {
+        HttpHeaders httpHeaders = (HttpHeaders) response.getNativeData(HTTP_HEADERS);
+        Type headersImpliedType = TypeUtils.getImpliedType(headersType);
+        if (headersImpliedType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+            headersImpliedType = TypeUtils.getReferredType(headersImpliedType);
+        }
+        Object headerMap;
+        if (headersImpliedType.getTag() == TypeTags.MAP_TAG) {
+            headerMap = createHeaderMap(httpHeaders,
+                    TypeUtils.getImpliedType(((MapType) headersImpliedType).getConstrainedType()));
+        } else if (headersImpliedType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            headerMap = createHeaderRecord(httpHeaders, (RecordType) headersImpliedType);
+        } else {
+            throw HttpUtil.createHttpError("unsupported headers type: " + headersType);
+        }
+        Object convertedHeaderMap = ValueUtils.convert(headerMap, headersType);
+        return validateConstraints(requireValidation, convertedHeaderMap, headersType, "header binding failed");
+    }
+
+    private static boolean hasHttpResponseType(Type targetType) {
+        return switch (targetType.getTag()) {
+            case TypeTags.OBJECT_TYPE_TAG -> true;
+            case TypeTags.UNION_TAG -> ((UnionType) targetType).getMemberTypes().stream().anyMatch(
+                    ExternResponseProcessor::hasHttpResponseType);
+            case TypeTags.TYPE_REFERENCED_TYPE_TAG -> hasHttpResponseType(TypeUtils.getImpliedType(targetType));
+            default -> false;
+        };
+    }
+
+    private static Optional<Type> getStatusCodeResponseType(Type targetType, String statusCode) {
+        if (isStatusCodeResponseType(targetType)) {
+            if (getStatusCode(targetType).equals(statusCode)) {
+                return Optional.of(targetType);
+            }
+        } else if (targetType instanceof UnionType unionType) {
+            List<Type> memberTypes = unionType.getMemberTypes();
+            for (Type memberType : memberTypes) {
+                Optional<Type> statusCodeResponseType = getStatusCodeResponseType(memberType, statusCode);
+                if (statusCodeResponseType.isPresent()) {
+                    return statusCodeResponseType;
+                }
+            }
```

Comment:
Can't we do something like this ?
```java
            return unionType.getMemberTypes().stream().map(member -> getStatusCodeResponseType(member, statusCode)).anyMatch(Optional::isPresent);
```

Reply:
You mean using a filter expression right?
```java
return unionType.getMemberTypes().stream().map(member -> getStatusCodeResponseType(member, statusCode)).
                    filter(Optional::isPresent).findFirst().orElse(Optional.empty());
```

Reply:
Changed like the above with [bce164f](https://github.com/ballerina-platform/module-ballerina-http/pull/1915/commits/bce164f18e1e33b548e2db2aa20f294e437a2202)

Reply:
Why can't we use `anyMatch` directly ?

Reply:
AnyMatch returns a boolean I need to find the first matching type

Reply:
Yes, my bad. I think in your approach we can remove `orElse` part as if there is no element `findFirst` will return an empty optional.

Reply:
Yeah but it will give an optional of the element type, here the map function make the element as Optional<Type> and the findFirst returns Optional<Optional<Type>>

Reply:
Fixed with [e0686d1](https://github.com/ballerina-platform/module-ballerina-http/pull/1915/commits/e0686d1349a324e03547df3ccea8e43e42fe8bdb)